linkis-computation-governance/linkis-manager/linkis-manager-persistence/src/main/java/org/apache/linkis/manager/persistence/impl/DefaultLabelManagerPersistence.java

```diff
@@ -113,15 +114,26 @@ public void removeLabel(PersistenceLabel persistenceLabel) {
 
     @Override
     public void updateLabel(int id, PersistenceLabel persistenceLabel) {
-        // 1.更新label表
-        // 2.删掉value
-        // 3.更新labelValue
-        // TODO: 2020/10/12  updateLabel 要重写判空
         persistenceLabel.setUpdateTime(new Date());
         labelManagerMapper.updateLabel(id, persistenceLabel);
         labelManagerMapper.deleteLabelKeyVaules(id);
         if (!persistenceLabel.getValue().isEmpty()) {
-            labelManagerMapper.registerLabelKeyValues(persistenceLabel.getValue(), id);
+            replaceIntoLabelKeyValues(persistenceLabel.getValue(), id);
+        }
+    }
+
+    private void replaceIntoLabelKeyValues(
+            Map<String, String> labelValueKeyAndContent, int labelId) {
+        if (null != labelValueKeyAndContent && labelId > 0) {
+            Iterator<Map.Entry<String, String>> iterator =
+                    labelValueKeyAndContent.entrySet().iterator();
+            while (iterator.hasNext()) {
```

Comment:
use foreach? like:
```
for(Map.Entry<String, String> labelKeyValue: labelValueKeyAndContent.entrySet())
```

Reply:
> use foreach? like:
> 
> ```
> for(Map.Entry<String, String> labelKeyValue: labelValueKeyAndContent.entrySet())
> ```

It is better to use iterator for map iteration, considering safety and performance

Reply:
got it, thanks.